ARTICLE.md

```diff
@@ -1,27 +1,290 @@
+background for any `Ingress`'s you define.
+
+#### HTTPS Made Easy
+
+In pursuit of our goal of end-to-end TLS for our web services,
+we will take advantage of what `Route`'s have to offer.
+Firstly, using a `Route` we can easily expose an HTTP `Service`
+with a hostname under a certain path:
+
+```
+---
+kind: Service
+apiVersion: v1
+metadata:
+  name: my-component-v1
+  namespace: my-app
+  labels:
+    app: my-app
+    component: my-component
+    version: v1
+spec:
+  type: ClusterIP
+  ports:
+    - port: 80
+      targetPort: http
+      protocol: TCP
+      name: http
+  selector:
+    app: my-app
+    component: my-component
+    version: v1
+---
+kind: Route
+apiVersion: v1
+metadata:
+  name: my-route
+  namespace: my-app
+  labels:
+    app: my-app
+    component: my-component
+spec:
+  host: my-component.apps.example.com
+  to:
+    name: my-component-v1
+    kind: Service
+  port:
+    targetPort: http
+```
+
+Even better, we can then secure our web service with edged
+terminated TLS and the ensure our users are always redirected
+to the HTTPS port of the load-balancer:
+
+```
+  tls:
+    termination: edge
+    insecureEdgeTerminationPolicy: Redirect
+```
+
+so now our `Route` as a whole looks like:
+
+```
+---
+kind: Route
+apiVersion: v1
+metadata:
+  name: my-route
+  namespace: my-app
+  labels:
+    app: my-app
+    component: my-component
+spec:
+  host: my-component.apps.example.com
+  to:
+    name: my-component-v1
+    kind: Service
+  port:
+    targetPort: http
+  tls:
+    termination: edge
+    insecureEdgeTerminationPolicy: Redirect
+```
+
+#### End-to-End TLS Made Easier
+
+Alright so that was pretty easy and we are a few steps closer
+to achieving end-to-end TLS. Moving forward, we first need our
+`Route` to no longer edge terminate TLS, which leaves us with
+two options for the termination policy: `passthrough` and
+`reencrypt`.
+
+With `passthrough`, the load-balancer will not participate in
+the TLS handshake and will simply send all traffic straight to
+the `Service` and then your `Pod`'s. That means your `Pod`'s
+would need a valid TLS cert for `my-component.apps.example.com`
+for the sake of the external client.
+
+With `reencrypt`, the load-balancer will handle the TLS
+handshake with the external client. Then it will do its own TLS
+handshake with the `Pod`'s at the end of your `Service`,
+re-encrypting the packets based on the TLS cert your pods are
+serving. Depending on what cert your `Pod`'s serve, you can
+specify a `destinationCACertificate` in the `Route` so that it
+trusts your cert. Run `oc explain route.spec.tls` for more
+details on all three policies and what additional options you
+have.
+
+Aside from needing a valid cert
+`my-component.apps.example.com`, the other drawback to
+`passthrough` is that internal `Service` hostname, i.e.
+`my-component-v1.my-app.svc.cluster.local`, would still be
+considered invalid. With `reencrypt` you could still achieve
```

Comment:
Other drawback to `passthrough` I need to mention is that path-based routing is not possible.